File: apps/ai-service/models/enums.py
from enum import Enum

class STATUS(str, Enum):
    PENDING = "pending"
    ACTIVE = "active"
    COMPLETED = "completed"
    ABANDONED = "abandoned"

class SessionType(str, Enum):
    AI_SESSION = "ai_session"
    HUMAN_SESSION = "human_session"

class TopicType(str, Enum):
    MOCK_INTERVIEW = "mock_interview"
    TECHNICAL = "technical"
    BEHAVIORAL = "behavioral"
    DEBATE = "debate"
    CUSTOM = "custom"

class RoleLevel(str, Enum):
    BEGINNER = "beginner"
    INTERMEDIATE = "intermediate"
    EXPERIENCED = "experienced"
    SENIOR = "senior"

class Difficulty(str, Enum):
    EASY = "easy"
    MEDIUM = "medium"
    HARD = "hard"
    EXPERT = "expert"
    
class AI_STRICTNESS(str, Enum):
    LENIENT = "lenient"
    BALANCED = "balanced"
    STRICT = "strict"
    ULTRA_STRICT ="ultra_strict"

class Domain(str, Enum):
    SWE = "swe"
    DATA_SCIENCE = "data_science"
    DEVOPS = "devops"
    PRODUCT = "product"
    CUSTOM = "custom"

class FileType(str, Enum):
    PDF = "pdf"
    DOCX = "docx"
    TXT = "txt"
    MD = "md"
    
class Speaker(str, Enum):
    USER = "user"
    AI = "ai"

class ContentType(str, Enum):
    QUESTION = "question"
    HINT = "hint"
    FEEDBACK = "feedback"
    SUMMARY = "summary"
    GREETING = "greeting"
    ANSWER = "answer"
    

File: apps/web/src/db/enums.ts
import { pgEnum } from "drizzle-orm/pg-core";

export const STATUS = [
    "pending",
    "active",
    "completed", 
    "abandoned",
] as const;

export const SESSION_TYPE = [
    "ai_session",
    "human_session",
] as const;

export const TOPIC_TYPE = [
    "mock_interview",
    "technical",
    "behavioral",
    "debate",
    "custom",
] as const;

export const ROLE_LEVEL = [
    "beginner",
    "intermediate",
    "experienced",
    "senior",
] as const;

export const DIFFICULTY = [
    "easy",
    "medium",
    "hard",
    "expert",
] as const;

export const DOMAIN = [
    "swe",
    "data_science",
    "devops",
    "product",
    "custom",
] as const;

export const AI_STRICTNESS = [
    "lenient",
    "balanced",
    "strict",
    "ultra_strict",
] as const;

export const FILE_TYPE = [
    "pdf",
    "docx",
    "txt",
    "md",
] as const;

export const SPEAKER = [
    "user",
    "ai",
] as const;

export const AI_RESPONSE_TYPE = [
    "question",
    "hint",
    "feedback",
    "summary",
    "greeting",
] as const;
export const CONTENT_TYPE = [
    ...AI_RESPONSE_TYPE,
    "answer",
] as const;

export const statusEnum = pgEnum("status", STATUS);
export const sessionTypeEnum = pgEnum("session_type", SESSION_TYPE);
export const topicTypeEnum = pgEnum("topic_type",TOPIC_TYPE);
export const roleLevelEnum = pgEnum("role_level",ROLE_LEVEL);
export const difficultyEnum = pgEnum("difficulty",DIFFICULTY);
export const domainEnum = pgEnum("domain",DOMAIN);
export const aiStrictnessEnum = pgEnum("ai_strictness",AI_STRICTNESS);
export const fileTypeEnum = pgEnum("file_type", FILE_TYPE);
export const speakerEnum = pgEnum("speaker", SPEAKER);
export const contentTypeEnum = pgEnum("content_type", CONTENT_TYPE);